src/index.ts

```diff
@@ -93,38 +216,204 @@ export class ConsentControl {
      * @param {ITextResources} textResources the text want to be displayed
      */
     public setTextResources(textResources: ITextResources): void {
-        if (textResources.bannerMessageHtml) {
-            this.textResources.bannerMessageHtml = textResources.bannerMessageHtml;
+        for (let key of Object.keys(this.textResources)) {
+            let typeKey = <keyof ITextResources> key;
+
+            if (textResources[typeKey]) {
+                this.textResources[typeKey] = textResources[typeKey];
+            }
+        }
+    }
+
+    /**
+     * Set themes in the banner and preferences dialog
+     * If a theme is not included, the default one will be used
+     * 
+     * @param {IThemes} themes themes that will be used in the banner and preferences dialog
+     */
+    public setThemes(themes?: IThemes): void {
```

Comment:
With this logic there's no way for defining a totally new theme. I would be able to set only light, dark, or high-contrast.

Instead of `setThemes()`, `setCertainTheme()` create:
`public createTheme(name: string, theme: ITheme)`
Which will create or override existing scheme.

And create
`public applyTheme(name: string)`
You have it below, but it's empty.

That should be the way how users work with the themes.